linetools/isgm/abssightline.py

```diff
@@ -0,0 +1,242 @@
+
+    Attributes
+    ----------
+    name : str
+        Name of the component, e.g. `Si II`
+    coord : SkyCoord
+        Sky coordinate
+    Zion : tuple 
+        Atomic number, ion -- (int,int)
+        e.g. (8,1) for OI
+    zcomp : float
+        Component redshift
+    vlim : Quantity array
+        Velocity limits of the component
+        e.g.  [-300,300]*u.km/u.s
+    A : int
+        Atomic mass -- used to distinguish isotopes
+    Ej : Quantity
+        Energy of lower level (1/cm)
+    comment : str
+        A comment, default is ``
+    """
+    __metaclass__ = ABCMeta
+
+    @classmethod
+    def from_abslines(cls, abslines, stars=None, **kwargs):
+        """Instantiate from a list of AbsLine objects
+
+        Parameters
+        ----------
+        abslines : list 
+          List of AbsLine objects
+        stars : str, optional
+          Asterisks to append to the ion name (e.g. fine-structure, CII*)
+        """
+        from .utils import build_components_from_abslines
+        # Check
+        if not isinstance(abslines, list):
+            raise IOError("Need a list of AbsLine objects")
+        if not all(isinstance(x, AbsLine) for x in abslines):
+            raise IOError("List needs to contain only AbsLine objects")
+
+        # Generate components
+        comps = build_components_from_abslines(abslines)
+
+        slf = cls.from_components(comps)
+
+        # Return
+        return slf
+
+    @classmethod
+    def from_components(cls, components, **kwargs):
+        """ Instantiate from a list of AbsComponent objects
+
+        Uses RA/DEC, Zion, Ej, A, z, vlim
+
+        Parameters
+        ----------
+        components : list
+           list of AbsComponent objects
+
+        Returns
+        -------
+        """
+        # Check
+        if not isinstance(components[0], AbsComponent):
+            raise IOError('Need an AbsComponent object')
+
+        # Instantiate with the first component
+        slf = cls( components[0].coord)
+        for comp in components:
+            slf.add_component(comp, **kwargs)
+        # Return
+        return slf
+
+    def __init__(self, radec, sl_type=None, em_type=None, comment=None, name=None):
+        """  Initiator
+
+        Parameters
+        ----------
+        radec : tuple or SkyCoord
+            (RA,DEC) in deg or astropy.coordinate
+        sl_type : str, optional
+          Sightline type, e.g. IGM
+        emtype : str, optional
+          Type of emission source for absorption sightline
+        comment : str, optional
+          A comment, default is ``
+        """
+        # Required
+        self.coord = ltu.radec_to_coord(radec)
+
+        # Lists
+        self._components = []
+
+        # Name
+        if name is None:
+            self.name = 'J{:s}{:s}'.format(
+                    self.coord.ra.to_string(unit=u.hour,sep='',pad=True),
+                    self.coord.dec.to_string(sep='',pad=True,alwayssign=True))
```

Comment:
Consider moving this naming code to a method called from linetools.utils, e.g.  `ltu.coord_to_str()` or similar.